Source/Core/Core/HW/WiimoteEmu/Extension/Shinkansen.cpp

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/HW/WiimoteEmu/Extension/Shinkansen.h"
+
+#include <array>
+
+#include "Common/Assert.h"
+#include "Common/Common.h"
+#include "Common/CommonTypes.h"
+#include "Core/HW/WiimoteEmu/WiimoteEmu.h"
+
+#include "InputCommon/ControllerEmu/ControlGroup/Buttons.h"
+#include "InputCommon/ControllerEmu/ControlGroup/MixedTriggers.h"
+
+namespace WiimoteEmu
+{
+// TODO: check this (only the last byte is known good)
+constexpr std::array<u8, 6> shinkansen_id{{0x00, 0x00, 0xa4, 0x20, 0x01, 0x10}};
+
+Shinkansen::Shinkansen() : Extension3rdParty("Shinkansen", _trans("Shinkansen Controller"))
+{
+  // Button layout on the controller:
+  //
+  //    Up      Select Start   D
+  // Left Right              A   C
+  //   Down                    B
+  //
+  groups.emplace_back(m_buttons = new ControllerEmu::Buttons(_trans("Buttons")));
+  m_buttons->AddInput(ControllerEmu::Translate, _trans("Left"));
+  m_buttons->AddInput(ControllerEmu::Translate, _trans("Down"));
+  m_buttons->AddInput(ControllerEmu::Translate, _trans("Right"));
+  m_buttons->AddInput(ControllerEmu::Translate, _trans("Up"));
+  m_buttons->AddInput(ControllerEmu::DoNotTranslate, _trans("A"));
+  m_buttons->AddInput(ControllerEmu::DoNotTranslate, _trans("B"));
+  m_buttons->AddInput(ControllerEmu::DoNotTranslate, _trans("C"));
+  m_buttons->AddInput(ControllerEmu::DoNotTranslate, _trans("D"));
```

Comment:
```suggestion
  m_buttons->AddInput(ControllerEmu::DoNotTranslate, "A");
  m_buttons->AddInput(ControllerEmu::DoNotTranslate, "B");
  m_buttons->AddInput(ControllerEmu::DoNotTranslate, "C");
  m_buttons->AddInput(ControllerEmu::DoNotTranslate, "D");
```

Reply:
I suppose this also applies to L/R below (but not L-Analog/R-Analog)?

Reply:
Since they have `ControllerEmu::Translate` set, and for consistency with the rest of the codebase, those should keep using `_trans`. Though, I'm not sure if there's a good reason why they're marked as translatable in the rest of the code base. (Probably because we started off by having everything translatable and me then making A/B/X/Y untranslatable to resolve an ambiguity between "B" the button and "B" the unit for bytes.)

Reply:
The levers on the controller are unlabeled (I think), so L/R are just English names to make it easier to tell which one is which.

Reply:
They are actually labeled if you look from above. But them being labeled doesn't necessarily have to mean the strings should be untranslatable. In particular, START/PAUSE is referred to using a localized name in official Japanese GameCube games, so I made sure we marked that one as translatable.

Reply:
Oops, I thought we were talking about GameCube controllers when I said that L and R are labeled...